pico_project.py

```diff
@@ -1128,127 +1128,134 @@ def generateProjectFiles(projectPath, projectName, sdkPath, projects, debugger):
 
     for p in projects :
         if p == 'vscode':
-            launch = ('{\n'
-                  '  "version": "0.2.0",\n'
-                  '  "configurations": [\n'
-                  '    {\n'
-                  '      "name": "Pico Debug (Cortex-Debug)",\n'
-                  '      "cwd": "${workspaceRoot}",\n'
-                  '      "executable": "${command:cmake.launchTargetPath}",\n'
-                  '      "request": "launch",\n'
-                  '      "type": "cortex-debug",\n'
-                  '      "servertype": "openocd",\n'
-                  f'      "gdbPath": "{gdbPath}",\n'
-                  '      "device": "RP2040",\n'
-                  '      "configFiles": [\n'
-                  f'        "interface/{debugger}",\n'
-                  '        "target/rp2040.cfg"\n'
-                  '        ],\n'
-                  '      "svdFile": "${env:PICO_SDK_PATH}/src/rp2040/hardware_regs/rp2040.svd",\n'
-                  '      "runToEntryPoint": "main",\n'
-                  '      // Give restart the same functionality as runToEntryPoint - main\n'
-                  '      "postRestartCommands": [\n'
-                  '          "break main",\n'
-                  '          "continue"\n'
-                  '      ],\n'
-                  '      "openOCDLaunchCommands": [\n'
-                  '          "adapter speed 1000"\n'
-                  '      ]\n'
-                  '    },\n'
-                  '    {\n'
-                  '      "name": "Pico Debug (Cortex-Debug with external OpenOCD)",\n'
-                  '      "cwd": "${workspaceRoot}",\n'
-                  '      "executable": "${command:cmake.launchTargetPath}",\n'
-                  '      "request": "launch",\n'
-                  '      "type": "cortex-debug",\n'
-                  '      "servertype": "external",\n'
-                  '      "gdbTarget": "localhost:3333",\n'
-                  f'      "gdbPath": "{gdbPath}",\n'
-                  '      "device": "RP2040",\n'
-                  '      "svdFile": "${env:PICO_SDK_PATH}/src/rp2040/hardware_regs/rp2040.svd",\n'
-                  '      "runToEntryPoint": "main",\n'
-                  '      // Give restart the same functionality as runToEntryPoint - main\n'
-                  '      "postRestartCommands": [\n'
-                  '          "break main",\n'
-                  '          "continue"\n'
-                  '      ]\n'
-                  '    },\n'
-                  '    {\n'
-                  '      "name": "Pico Debug (C++ Debugger)",\n'
-                  '      "type": "cppdbg",\n'
-                  '      "request": "launch",\n'
-                  '      "cwd": "${workspaceRoot}",\n'
-                  '      "program": "${command:cmake.launchTargetPath}",\n'
-                  '      "MIMode": "gdb",\n'
-                  '      "miDebuggerPath": "{gdbPath}",\n'
```

Comment:
Huh, looks like there was a hidden bug here, where the `f` was missing from the start of the string :wink:

Reply:
I seems so..., but

Resolved 👌